### PR TITLE
Add execution mutation to GQL schema

### PR DESCRIPTION
**Pull request by LucasPickering**
Adds some minimal API shit to allow executing programs server side. This will execute the program and, if successful, store statistical perf data on it. The API isn't fully fleshed out, but all the data we need is being stored. The next step is to add more fields into the API output that we have (hence all the TODOs). After that we can add some more API stuff to actually read leaderboard data, then after that we can do UI.

### CODE REVIEW

File: api/schema.graphql [hunk 1]
```diff
@@ -850,3 +864,96 @@ type DeleteUserProgramPayload {
...
+Execution output when a program compiles and executes without error, but the
+final state doesn't match the expected state (as defined by the program spec).
+"""
+type ProgramFailureOutput {
+  # TODO add machine
+  """
+  TODO
+  """
+  todo: Int! @juniper(infallible: true, ownership: "owned")
+}
```
Comment (JRMurr) [inline api/schema.graphql:920]: maybe `programRejected` or denied?

**Review by JRMurr — CHANGES_REQUESTED** [review]

File: api/schema.graphql [hunk 2]
```diff
@@ -952,7 +952,7 @@ type ExecuteUserProgramPayload {
   outcome, which could be one of:
   - Compile error
   - Runtime error
-  - Failed execution (incorrect input/output at termination)
+  - Rejected execution (incorrect input/output at termination)
   - Success
```
Comment (JRMurr) [inline api/schema.graphql:956]: Really minor but missed accepted here

**Review by JRMurr — CHANGES_REQUESTED** [review]

File: api/schema.graphql (hunk 2 above)
Comment (LucasPickering) [reply api/schema.graphql:956]: Yeah ik I was just gonna leave that one cause I don't think it makes a diff

Comment (JRMurr) [reply api/schema.graphql:956]: good enough for me lol

**Review by JRMurr — APPROVED** [review]

**Pull request merged**